docs/guides/integration/jupyter.md

```diff
@@ -0,0 +1,45 @@
+# Using uv with Jupyter
+
+## As a standalone tool...
+
+If you're working within a [project](../../concepts/projects.md), you can kick off a Jupyter server
+with access to the project's virtual environment via the following:
+
+```console
+$ uv run --with jupyter jupyter notebook
+```
+
+Within the notebook, you can then import your project's modules as you would in any other `uv run`
+invocation. For example, if your project depends on `requests`, `import requests` will import
+`requests` from the project's virtual environment.
+
+While `jupyter` itself is installed in an isolated environment when used via
+`uv run --with jupyter`, within the notebook, `!uv add` and related commands will modify the
+_project's_ environment.
+
+For example, running `!uv add pydantic` from within a notebook will add `pydantic` to the project's
+dependencies and virtual environment, such that `import pydantic` will work immediately, without
+further configuration or a server restart.
+
+!!! note
+
+    Since the Jupyter server is running in an isolated virtual environment, `!uv pip install` will install package's
+    into _Jupyter's_ environment, not the project environment. Such dependencies may disappear on subsequent `jupyter`
+    invocations. To install packages into the project environment, use `!uv add`.
```

Comment:
Actually, it looks like this is solvable using kernel `ipython kernel install` (https://github.com/astral-sh/uv/issues/6329#issuecomment-2302252543).